hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -223,6 +226,11 @@ public static void verifyKeyName(String keyName) {
     if (keyName == null) {
       throw new IllegalArgumentException("Key name is null");
     }
+    if (keyName.startsWith(OM_SNAPSHOT_INDICATOR + OM_KEY_PREFIX)) {
```

Comment:
I think you have to move out this to separate check because [validateKeyName](https://github.com/apache/ozone/blob/3dd6e29f2deb1d27aef2e82fcf9ef962e996cc03/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java#L596) only runs when [checkKeyNameEnabled](https://github.com/apache/ozone/blob/e9592d2654e86ebc64648195e1578578eda3f389/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequest.java#L100).

Reply:
yup! moved to another separated method in OmUtils.